src/app/qgswelcomepage.cpp

```diff
@@ -45,38 +47,52 @@ QgsWelcomePage::QgsWelcomePage( bool skipVersionCheck, QWidget *parent )
+  QLabel *recentProjectsTitle = new QLabel( QStringLiteral( "<div style='font-size:%1px;font-weight:bold'>%2</div>" ).arg( QString::number( titleSize ), tr( "Recent Projects" ) ) );
   recentProjectsTitle->setContentsMargins( titleSize / 2, titleSize / 6, 0, 0 );
-
-  recentProjectsContainer->layout()->addWidget( recentProjectsTitle );
+  centerLayout->addWidget( recentProjectsTitle, 0, 0 );
 
   mRecentProjectsListView = new QListView();
   mRecentProjectsListView->setResizeMode( QListView::Adjust );
   mRecentProjectsListView->setContextMenuPolicy( Qt::CustomContextMenu );
   connect( mRecentProjectsListView, &QListView::customContextMenuRequested, this, &QgsWelcomePage::showContextMenuForProjects );
 
-  mModel = new QgsWelcomePageItemsModel( mRecentProjectsListView );
-  mRecentProjectsListView->setModel( mModel );
-  mRecentProjectsListView->setItemDelegate( new QgsWelcomePageItemDelegate( mRecentProjectsListView ) );
+  mRecentProjectsModel = new QgsRecentProjectItemsModel( mRecentProjectsListView );
+  mRecentProjectsListView->setModel( mRecentProjectsModel );
+  mRecentProjectsListView->setItemDelegate( new QgsRecentProjectItemDelegate( mRecentProjectsListView ) );
+
+  centerLayout->addWidget( mRecentProjectsListView, 1, 0 );
 
-  recentProjectsContainer->layout()->addWidget( mRecentProjectsListView );
+  layout->addWidget( centerContainer );
 
-  layout->addWidget( recentProjectsContainer );
+  QLabel *templatesTitle = new QLabel( QStringLiteral( "<div style='font-size:%1px;font-weight:bold'>%2</div>" ).arg( titleSize ).arg( tr( "Templates" ) ) );
+  templatesTitle->setContentsMargins( titleSize / 2, titleSize / 6, 0, 0 );
+  centerLayout->addWidget( templatesTitle, 0, 1 );
+
+  mTemplateProjectsModel = new QgsTemplateProjectsModel( this );
+  mTemplateProjectsListView = new QListView();
+  mTemplateProjectsListView->setResizeMode( QListView::Adjust );
+  mTemplateProjectsListView->setModel( mTemplateProjectsModel );
+  mTemplateProjectsListView->setItemDelegate( new QgsRecentProjectItemDelegate( mTemplateProjectsListView ) );
+  mTemplateProjectsListView->setContextMenuPolicy( Qt::CustomContextMenu );
+  connect( mTemplateProjectsListView, &QListView::customContextMenuRequested, this, &QgsWelcomePage::showContextMenuForTemplates );
+  centerLayout->addWidget( mTemplateProjectsListView, 1, 1 );
 
   mVersionInformation = new QTextBrowser;
   mVersionInformation->setSizePolicy( QSizePolicy::Expanding, QSizePolicy::Maximum );
   mVersionInformation->setReadOnly( true );
   mVersionInformation->setOpenExternalLinks( true );
-  mVersionInformation->setStyleSheet( "QTextEdit { background-color: #dff0d8; border: 1px solid #8e998a; padding-top: 0.25em; max-height: 1.75em; min-height: 1.75em; } "
+  mVersionInformation->setStyleSheet( QStringLiteral( "QTextEdit { background-color: #dff0d8; border: 1px solid #8e998a; padding-top: 0.25em; max-height: 1.75em; min-height: 1.75em; } "
```

Comment:
@nirvn are these hardcoded colors theme safe?

Reply:
@nyalldawson , yep, that's fine (in fact, this is the result of a fix that made the new version info bar look good across all themes).